lib/ModelMixins/GroupMixin.ts

```diff
@@ -108,17 +108,29 @@ function GroupMixin<T extends Constructor<Model<GroupTraits>>>(Base: T) {
        * - member.uniqueId = 'some-group-id/some-member-id'
        * - group.shareKeys = 'old-group-id'
        * - So we want to create member.shareKeys = ["old-group-id/some-member-id"]
+       *
+       * We also repeat this process for each shareKey for each member
        */
 
       this.memberModels.forEach((model: BaseModel) => {
         // Only add shareKey if model.uniqueId is an autoID (i.e. contains groupId)
         if (isDefined(model.uniqueId) && model.uniqueId.includes(groupId)) {
-          shareKeys.forEach(groupShareKey =>
+          shareKeys.forEach(groupShareKey => {
+            // Get shareKeys for current model
+            const modelShareKeys = this.terria.modelIdShareKeysMap.get(
+              model.uniqueId!
+            );
+            modelShareKeys?.forEach(modelShareKey => {
+              this.terria.addShareKey(
+                model.uniqueId!,
+                modelShareKey.replace(groupId, groupShareKey)
+              );
+            });
             this.terria.addShareKey(
               model.uniqueId!,
               model.uniqueId!.replace(groupId, groupShareKey)
-            )
-          );
+            );
+          });
```

Comment:
This adds another level of `shareKey` generation. 
This means that if the group has `shareKeys` and a member has `shareKeys` - new `shareKeys` will be added by replacing the `groupId` element of each `shareKey` with each group `shareKey

For example

- We have a WMS group with id `"/groupID"` and sharekeys `["groupShareID"]`
- This group has a WMS item with id `"/groupID/itemID"` and sharekeys `["groupID/itemShareID"]`

With this change, the WMS item will now get an extra `shareKey` 
- `["groupID/itemShareID", "groupShareID/itemShareID"]`

This is generated by replacing the group's ID (`groupID`) with the groups shareKey (`groupShareID`)